pkg/gardenlet/controller/shoot/care/reconciler.go

```diff
@@ -118,7 +119,8 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		shoot,
 	)
 	if err != nil {
-		if err := r.patchStatusToUnknown(ctx, shoot, "Precondition failed: operation could not be initialized", shootConditions.ConvertToSlice(), shootConstraints.ConvertToSlice()); err != nil {
+		updatedConditions, updatedConstraints := r.setStatusToUnknown("Precondition failed: operation could not be initialized", shootConditions.ConvertToSlice(), shootConstraints.ConvertToSlice())
+		if err := r.patchStatus(ctx, log, shoot, shootConditions, updatedConditions, shootConstraints, updatedConstraints); err != nil {
```

Comment:
Can you update the [corresponding unit tests](https://github.com/gardener/gardener/blob/d82db8674c240691f9276c6775dc04e573bb8c05/pkg/gardenlet/controller/shoot/care/reconciler_test.go#L126) to verify that we no longer removed other conditions/constraints?